### PR TITLE
Todo settings

### PR DESCRIPTION
**Pull request by tmat**
Replaces CommentTaskTokenSerializer specialized option provider with standard RoamingProfileStorageLocation storage that reads the options using `ISettingManager`. 

There is an oddity with how this option is stored in the Settings Manager. If the option is read as `object` the Settings Manager returns Newtonsoft.Json's JArray object. To avoid taking dependency on this type we explicitly query for `string[]`. If we do so we get back `string[]` that we can use directly.

Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1585190

### CODE REVIEW

**Comment by tmat** [comment]
@jasonmalinowski PTAL

File: src/Features/Core/Portable/TodoComments/TodoCommentDescriptor.cs [hunk 1]
```diff
@@ -24,28 +26,18 @@ public TodoCommentDescriptor(string text, int priority)
...
         }
 
-        public static ImmutableArray<TodoCommentDescriptor> Parse(string data)
+        public static ImmutableArray<TodoCommentDescriptor> Parse(ImmutableArray<string> items)
         {
-            if (string.IsNullOrWhiteSpace(data))
-                return ImmutableArray<TodoCommentDescriptor>.Empty;
-
-            var tuples = data.Split('|');
             using var _ = ArrayBuilder<TodoCommentDescriptor>.GetInstance(out var result);
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/TodoComments/TodoCommentDescriptor.cs:31]: Nit, but we can we pass capacity? (This probaby runs rarely enough to matter though.)

File: src/Features/Core/Portable/TodoComments/TodoCommentDescriptor.cs [hunk 2]
```diff
@@ -24,28 +26,18 @@ public TodoCommentDescriptor(string text, int priority)
...
-                var pair = tuple.Split(':');
-
-                if (pair.Length != 2 || string.IsNullOrWhiteSpace(pair[0]))
-                    continue;
-
-                if (!int.TryParse(pair[1], NumberStyles.None, CultureInfo.InvariantCulture, out var priority))
-                    continue;
-
-                result.Add(new TodoCommentDescriptor(pair[0].Trim(), priority));
+                if (item.Split(':') is [var token, var priorityString] &&
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/TodoComments/TodoCommentDescriptor.cs:35]: Ooh that's pretty.

File: src/Features/Core/Portable/TodoComments/TodoCommentDescriptor.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/Features/Core/Portable/TodoComments/TodoCommentDescriptor.cs:31]: (Feel free to ignore unless you have to make some other edit.)

File: src/Features/LanguageServer/Protocol/Features/TodoComments/TodoCommentOptionsStorage.cs [hunk 3]
```diff
@@ -2,15 +2,24 @@
...
+using System.Collections.Generic;
+using System.Collections.Immutable;
 using Microsoft.CodeAnalysis.Options;
 
 namespace Microsoft.CodeAnalysis.TodoComments
 {
     internal static class TodoCommentOptionsStorage
     {
-        public static readonly Option2<string> TokenList = new("TodoCommentOptions", "TokenList", TodoCommentOptions.Default.TokenList);
+        public static readonly Option2<ImmutableArray<string>> TokenList = new(
```
Comment (jasonmalinowski) [inline src/Features/LanguageServer/Protocol/Features/TodoComments/TodoCommentOptionsStorage.cs:13]: We don't have any way to say the underlying storage is an ImmutableArray<> but expose it as the nicer types? That way we could avoid the caching in the AbstractTodoCommentsIncrementalAnalyzer?

**Review by jasonmalinowski — APPROVED** [review]

File: src/Features/LanguageServer/Protocol/Features/TodoComments/TodoCommentOptionsStorage.cs (hunk 3 above)
Comment (tmat) [reply src/Features/LanguageServer/Protocol/Features/TodoComments/TodoCommentOptionsStorage.cs:13]: Not without introducing a custom option storage. We can do that later.

File: src/Features/Core/Portable/TodoComments/TodoCommentDescriptor.cs [hunk 4]
```diff
@@ -24,28 +26,18 @@ public TodoCommentDescriptor(string text, int priority)
             Priority = priority;
         }
 
-        public static ImmutableArray<TodoCommentDescriptor> Parse(string data)
+        public static ImmutableArray<TodoCommentDescriptor> Parse(ImmutableArray<string> items)
```
Comment (davidwengier) [inline src/Features/Core/Portable/TodoComments/TodoCommentDescriptor.cs:29]: https://github.com/xamarin/vsmac/pull/8604 would indicated that TypeScript has IVT usage of this. Will try to confirm

**Pull request merged**